### ISSUE TITLE
Derive Clone + Debug for descriptor structs

### ISSUE THREAD

**Issue by kvark** [retep998/winapi-rs#542]
Things like `D3D11_RASTERIZER_DESC` we used to print to debug output in `gfx_device_d3d11`...

**Comment by retep998** [comment]
Add `#[debug]` to the specific structs that you need Debug impls for and submit a PR.

**Comment by tangmi** [comment]
I've made an attempt in #760, but am unsure how to proceed with e.g. `DXGI_ADAPTER_DESC::Description`, which is a `[WCHAR; 128]` and doesn't have a derivable `Debug`. Ideally we can make a custom override that is something like `&CStr::from_ptr(&self.Description)` (and also for things like `D3D11_INPUT_ELEMENT_DESC::SemanticName`, which is `LPCSTR`), but I'm unsure how "custom" `winapi-rs` wants to be.

**Comment by retep998** [comment]
For types which cannot derive `Debug`, they just won't implement `Debug` until someone gets const generics implemented and stabilized so that they can derive `Debug`.

**Comment by tangmi** [comment]
I think the best thing to do is remove `#[debug]` from `DXGI_ADAPTER_DESC*` and create a new issue/pr adding them that's blocked on const generics. Thoughts?
